### ISSUE TITLE
ngAnimate not work

### ISSUE THREAD

**Issue by furyscript** [angular/angular.js#12345]
I have this container
```html
<div ui-view="view-layout" autoscroll="false" class="wrapper"></div>
```
I use ui-route to change state / route.

In this ui-view I attach this
```html
<header ng-include="'common/nav/top-navbar/top-navbar.tpl.html'" ng-controller="TopnavbarController" class="topnavbar-wrapper"></header>
<aside ng-include="'common/nav/sidebar/sidebar.tpl.html'" ng-controller="SidebarController" class="aside"></aside>
<aside ng-include="'common/nav/offsidebar/offsidebar.tpl.html'" class="offsidebar"></aside>

<section>
    <!-- Page content animated fadeInUp-->
    <div ui-view="view-app" autoscroll="false" class="content-wrapper ng-fadeInUp"></div>
    
</section>
<footer ng-include="'common/nav/footer.tpl.html'"></footer>
```
The section is dynamic based on page content. The problem is that the animation fadeInUp is binded to "div wrapper" and not in "div content-wrapper" where I want it. Any solution??

**Comment by petebacondarwin** [comment]
I think we need some  more information about this issue before we can help you. For instance the CSS that defines the animation. Best of all, if you could provide a working demonstration of the problem you would be more likely to get a solution.

**Comment by furyscript** [comment]
The css works fine, but is attached to a parent div, I cannot make demostration...

**Comment by petebacondarwin** [comment]
In that case we are not able to help.

**Comment by furyscript** [comment]
http://plnkr.co/edit/VxnHAeuZ9JmmallbLVsx
This is plunkr @petebacondarwin

**Comment by petebacondarwin** [comment]
OK I will take a look this week

**Comment by furyscript** [comment]
Thank you, I need your help before as you can :)

**Comment by petebacondarwin** [comment]
So what I see in that plunker is that you have attached the animations to the nested view by using the `[ui-view]` selector in the CSS and then using `data-ui-view` on the wrapper view placeholder to prevent it being animated. Right?

I think that the problem is that ui-router is loading the nested view into the wrapper view before the wrapper view is added to the DOM and so the `ngAnimate` code is not picking up the `enter` correctly.
I am not sure if this is a problem with `ngAnimate` or with `ui-router`.

@matsko, @nateabele or @christopherthielen do you have a view on this?

**Comment by petebacondarwin** [comment]
I also note that this plunker is using angular 1.2.x - have you tried updating to 1.4?

**Comment by furyscript** [comment]
@petebacondarwin  Yes it's right :)
I'm trying with 1.4.x(2-3) but also not work.
Have you a stable fix for this or I should waiting?

**Comment by matsko** [comment]
I found the issue. You're using `data-ui-view` in your HTML code, but your CSS selectors are only matching [ui-view]. Make sure that the two are the same:
http://plnkr.co/edit/2VSZtYsjQ2p7lXgVbFlw?p=preview

**Comment by furyscript** [comment]
Isn't correct. I need to make animation in "view-app" and not to div that contain it ("view-layout")
@matsko

**Comment by nateabele** [comment]
@petebacondarwin Our animation code is a little hard to follow because our Angular support goes back a few versions, but this is basically what we do:

 - https://github.com/angular-ui/ui-router/blob/master/src/viewDirective.js#L219-L223
 - https://github.com/angular-ui/ui-router/blob/master/src/viewDirective.js#L270-L271

If we need to change up the priority on something, let us know.

**Comment by furyscript** [comment]
I wait other comments, maybe somebody know solutions...
If you see http://themicon.co/theme/angle/v2.6.1/backend-angular/#/app/dashboard I see that the change nesteded state works fine, but in my plunkr not... I take a look a few days I their example but I don't find difference... Maybe is lazyload library?? @petebacondarwin @nateabele @matsko

**Comment by furyscript** [comment]
Nobody?

**Comment by furyscript** [comment]
I need help!

**Comment by matsko** [comment]
Sorry for the delay. Would you mind testing against these files:
http://www.angular-dev.com/raf-fix/angular.js
http://www.angular-dev.com/raf-fix/angular-animate.js

If not then we can pair tomorrow on the issue via hangouts.

**Comment by furyscript** [comment]
@matsko still not working,
see http://plnkr.co/edit/HbOueeaqRZF6IjNK8PPR?p=preview
Thanks for reply I need fix as soon as possible, thanks!

**Comment by matsko** [comment]
OK I'll have a look at this tomorrow.

**Comment by furyscript** [comment]
Thank you very much! @matsko

**Comment by savras** [comment]
@furyscript 
I've updated animate.css from your latest plunk:
http://plnkr.co/edit/EbZVmWugk9UmlENKTfBA?p=preview

Its more like a workaround. Would that do?

**Comment by furyscript** [comment]
Uhm I've a large CSS with all animations adpted see https://daneden.github.io/animate.css/
It's probably impossible or it isn't good updated all css D: @savras 
I prefer a really update in code..

**Comment by furyscript** [comment]
And is it possible with this example? http://plnkr.co/edit/m2Cil3QHONx06egtW2Ou?p=preview

**Comment by furyscript** [comment]
HELP!

**Comment by furyscript** [comment]
No solutions? D:

**Comment by furyscript** [comment]
Hi all !!
I'm really happy I've found solution :D 
check this plunker: http://plnkr.co/edit/2b2QvcbekC79HnwnbTRB?p=preview
@petebacondarwin @matsko @savras @nateabele

**Comment by Narretz** [comment]
@furyscript Glad that you found a solution. I think this behavior can be attributed more to ui-view than to angular-animate. since it's fixable by changing the API (and if it was a bug with ng-animate I'd expect it to have popped up more often).

**Comment by nateabele** [comment]
@Narretz The DOM implementation for `ui-view` is a copy of `ng-view`, so, not likely, unless it's also an `ng-view` problem.
